### ISSUE TITLE
Since upgrade, Function reading from EventHub batches now reading triple

### ISSUE THREAD

**Issue by stephlocke** [Azure/azure-webjobs-sdk#890]
I upgraded my Functions this morning. My Function that reads from a batch usually causes my EventHub outputs to be *double* the number of inputs. Since this morning that figure is *triple* the number of inputs and my database which is receiving the data has gone to 100% cpu suggesting it's now working pretty darn hard.

I'm using the 2.0 beta of the service bus pkg to get batch eventhub reads. I'm now very worried it always over processed and is now crazily over-processing.

It's super hard to reconcile an eventhub, especially to a DB that's basically down due to the volumes it's ingesting. I can't get any info from the monitoring page because none of the function runs are showing on the page. 

It **feels** like a bug caused by the upgrade. Any ideas on how to get more info on this?

```
#load "..\shared\model.csx"

#r "Newtonsoft.Json"
using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Microsoft.Azure.WebJobs.ServiceBus;
using Microsoft.ServiceBus.Messaging;

public static void Run([EventHubTrigger("webhits")] Webhit[] myEventHubMessage, TraceWriter log)
{
    foreach(Webhit webhit in myEventHubMessage)
    {
       //do something
    }
}
```

![image](https://cloud.githubusercontent.com/assets/6452610/19811995/a778c08c-9d2b-11e6-8c37-9ae97a8a2915.png)

**Comment by MikeStall** [comment]
Can you clarify: what do incoming and outgoing messages mean here?  Are you saying that an event is being processed multiple times?

**Comment by stephlocke** [comment]
This is the primary EventHub metric that I use to get an idea of load - incoming are the ones that get dropped into EventHubs and the outputs are the reads. I'd assumed it I had double outputs because I was reading it in a function and also getting a read when it was being put into the EventHub Archives.

**Comment by stephlocke** [comment]
@MikeStall I can see a 0.8 version - should I upgrade?

**Comment by davidebbo** [comment]
Yes, please do upgrade to be on latest. Although I would not expect it to affect this particular behavior.

**Comment by stephlocke** [comment]
Upgraded - situation on eventhub remains the same

**Comment by stephlocke** [comment]
It looks like this behaviour stopped around midnight. 

I hope there was some action taken by someone somewhere that resolved this. I don't mind problems, but I don't like transient problems that aren't explainable! :(
![image](https://cloud.githubusercontent.com/assets/6452610/19828252/64cdbb14-9db9-11e6-9ca3-eb24e621e9fe.png)

**Comment by lindydonna** [comment]
Assigning to @MikeStall for investigation

**Comment by mathewc** [comment]
@stephlocke Is this still an issue?

**Comment by apollak** [comment]
@mathewc I noticed the same issue, having twice as many outgoing messages instead of tripple like stephlocke. I am using a custom consumer group. First I thought it might be linked to the amount of consumer groups, but adding a third consumer group did not tripple my messages.

Data is passed as JSON-Array to Eventhub and processed in an Azure Function which passes on the messages in the Array to DocumentDB. Here is the Azure Function Code:

```csharp
#r "Newtonsoft.Json"

using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Configuration;

public static void Run(string myEventHubMessage, ICollector<JObject> outputDocument, TraceWriter log)
{    
    log.Info($"Incomming document: {myEventHubMessage}");
    JArray inputDataArray = JArray.Parse(myEventHubMessage) as JArray;
    dynamic inputData = inputDataArray;
    foreach(dynamic data in inputData)
    {
        string jsonSingleOutput = JsonConvert.SerializeObject(data);
        outputDocument.Add(data);
        log.Info($"Sent document: {jsonSingleOutput}");            
    }
}

**Comment by apollak** [comment]
@mathewc I noticed the same issue, having twice as many outgoing messages instead of tripple like stephlocke. I am not using the SDK though but added the function directly into the portal. Maybe the issue is linked to this?

I am using a custom consumer group. First I thought it might be linked to the amount of consumer groups, but adding a third consumer group did not tripple my messages.

Data is passed as JSON-Array to Eventhub and processed in an Azure Function which passes on the messages in the Array to DocumentDB. Here is the Azure Function Code:

```csharp
#r "Newtonsoft.Json"

using System;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Configuration;

public static void Run(string myEventHubMessage, ICollector<JObject> outputDocument, TraceWriter log)
{    
    log.Info($"Incomming document: {myEventHubMessage}");
    JArray inputDataArray = JArray.Parse(myEventHubMessage) as JArray;
    dynamic inputData = inputDataArray;
    foreach(dynamic data in inputData)
    {
        string jsonSingleOutput = JsonConvert.SerializeObject(data);
        outputDocument.Add(data);
        log.Info($"Sent document: {jsonSingleOutput}");            
    }
}

**Comment by saldistefano** [comment]
I see this same behavior. We thought it was related to Consumer Groups but does not seem to change regardless of number of consumer groups.
